compiler-lib/src/debugging.rs

```diff
@@ -63,36 +63,108 @@ use std::{
     thread::{self, JoinHandle},
 };
 
-#[cfg(feature = "debugger_gui")]
+#[cfg(any(feature = "debugger_gui", feature = "debugger_vscode"))]
```

Comment:
since both breakpoint implementations have nothing in common, why not make a seprate #[cfg(feature = "debugger_vscode")] implementation?

Reply:
They have a lot in common. This way, you can use the debugger server and I can use the debugger visualizer in vs code without needing to change anything or having code duplicates.

Reply:
What I mean is this:

```
#[cfg(feature = "debugger_gui")]
macro_rules! breakpoint {
// all the stuff from the if-truee-branches in the current impl
}
#[cfg(feature = "debugger_vscode")]
macro_rules! breakpoint {
// all the stuff from the else-branches in the current impl
}
```